dev: add opt-in pre-commit hooks (#2780)
opendp/opendp: docs/source/contributing/development-environment.rst

* Be sure to install the Python and Rust plugins for interactivity
* Be sure to "attach" the Cargo.toml in the red banner the first time you open a Rust source file
* Use run configurations to `build the Rust library <https://plugins.jetbrains.com/plugin/8182-rust/docs/cargo-command-configuration.html#cargo-command-config>`_ and run tests

> Git Hooks

Reviewer: ```suggestion
Git Hooks
---------
```
RST formatting

A suggested (opt-in) pre-commit hook lives in ``.githooks``.

> Enable it with ``git config core.hooksPath .githooks``; see ``.githooks/README.md`` for details.

Reviewer: If details are needed, suggest a link to the file on github, instead of just the filename.

Git only runs the hook if it is executable. A fresh clone is fine, but if the executable
bit was stripped, git will skip the hook or complain it is not executable.

> Restore it with ``chmod +x .githooks/pre-commit``

Reviewer: duplicate. Once at most?

* Be sure to install the Python and Rust plugins for interactivity
* Be sure to "attach" the Cargo.toml in the red banner the first time you open a Rust source file
* Use run configurations to `build the Rust library <https://plugins.jetbrains.com/plugin/8182-rust/docs/cargo-command-configuration.html#cargo-command-config>`_ and run tests

Git Hooks
---------

A suggested (opt-in) pre-commit hook lives in ``.hooks-suggested``.

> Enable it with ``git config core.hooksPath .hooks-suggested``.

Reviewer: ```suggestion
Enable it with ``cp .hooks-suggested/pre-commit .git/hooks/``
```
If we are thinking of the pre-commits as something that is left to the preference of individual developers, slight preference for `cp`. But if we want to encourage the same patterns across the team, this is better. 

Decided I don't really care.

`See <https://github.com/opendp/opendp/tree/main/.hooks-suggested/README.md>` for more details.

> md

Reviewer: ```suggestion
```
Stray line?
